src/options.js

```diff
@@ -288,7 +288,7 @@ export function maybeInterval(interval, type) {
   if (typeof interval === "string") return (type === "time" ? maybeTimeInterval : maybeUtcInterval)(interval);
   if (typeof interval.floor !== "function") throw new Error("invalid interval; missing floor method");
   if (typeof interval.offset !== "function") throw new Error("invalid interval; missing offset method");
-  return interval;
+  return maybeIsoInterval(interval);
```

Comment:
Since `maybeIsoInterval` is only used here, I suggest we inline.

```suggestion
  return isTimeInterval(interval) ? isoInterval(interval) : interval;
```